In [7]:
from google_play_scraper import Sort, reviews_all
import pandas as pd
import numpy as np



In [2]:
us_reviews = reviews_all(
    'com.spotify.music', # the id is found within the app url at play.google.com/store
    sleep_milliseconds=0, 
    lang='en', 
    country='us', 
    sort=Sort.MOST_RELEVANT, # defaults to Sort.MOST_RELEVANT
    filter_score_with=5 # defaults to None(means all score)
)

In [83]:
df_spotify = pd.DataFrame(np.array(us_reviews),columns=['review'])


df_spotify = df_spotify.join(pd.DataFrame(df_spotify.pop('review').tolist()))

df_spotify.head()




reviewId  \
0  fa253801-1fe3-4b33-aaae-809979d09ea4   
1  5cf3d50b-64e0-4271-ac65-b7c3dc0a8319   
2  4abd54b6-93d7-4995-a9b4-b390954457fa   
3  bb2a96e7-3f38-4007-96b6-c83f010d5de2   
4  e8afbe49-6e6a-4fe1-9aa4-032f1be42843   

                                    userName  \
0                               Angela Kelly   
1                               Arielle Tann   
2                            Trenton Piersma   
3                              Rachel Belden   
4  Im an idiot so just ignore the following:   

                                                                                 userImage  \
0  https://play-lh.googleusercontent.com/a-/AFdZucob7UpxxhKjI2475A-3ry88NSTQ41uc3tqN2Sc7zA   
1  https://play-lh.googleusercontent.com/a-/AFdZucrGjHWPEt2_zQRfckEPNkURu2pmk3ECLJPZLaZU9g   
2  https://play-lh.googleusercontent.com/a/AItbvmleN3oqrSmUcYC6gX4Kr5FGsJ6AMjoDRw2x0C0H=mo   
3  https://play-lh.googleusercontent.com/a-/AFdZucpYTmVBEFIqnE-g3xMy00SGS8z0YQYZYiqzAzqMyA   
4  https://play-lh.googleusercontent.com/a-/AFdZucqlo0q_sRzAR50GWAfQEkK1Vp3uAE9WumX0Kt-XSg   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              content  \
0                                                                                                                                          I enjoy using the app and have all of my music, podcasts and some audiobooks all in one place. I wish there was a better way to organize files a little more specific then what is available now but I make it work. Quality has always been good! The one or two times it messed up after an update I cleared cache, forced stop and the reopened the app. Simple enough!   
1                                                                                                                                                                   Napster has gone to the dogs, so I finally bit the bullet and went to Spotify. Never been happier! Love the non-delay between tracks, podcasts, the enhance and radio features, and most of all, the reliability, the fact that it just WORKS! And to top it all off, I got 3 months free because I have a Samsung phone. Keep up the great work!   
2    Hands down the best app to listen to music with. Cheap Monthly payments for virtually any song you could want. Being able to create playlists of any songs is a very powerful tool to have. And the ability to track your stats with another app is a very immersive experience that you wouldn't really think is useful. But it's really cool to be able to see what songs/artists you've listened to the most. Alot of people complain its buggy but out of the 5 years I've been using spotify they're minor.   
3                                                                                                                            With both the free version and the upgrade to the paid version; I have NEVER been disappointed in my music options. Spotify almost always has artists that I'm looking for and if I don't know exactly what I want to listen to, I pick the radio option to the artist that best fits my mood and then I get to discover new music. I love Spotify and totally recommend it to everyone!   
4  Love it! Switched from Amazon music to Spotify after having my playlists on Amazon completely erased after I changed my password. Thousands of downloaded songs gone just like that, Amazon customer service said they couldn't do anything about it, to just go ahead and download them again, ha! Not doing that again. Love the feature on spotify that allows you to start the next song before th

In [84]:
df_spotify.drop(columns={'reviewId', 'userImage', 'reviewCreatedVersion'}, inplace=True )
df_spotify = df_spotify.rename(columns={'userName':'user', 'content':'review_description', 'score':'rating', 'at':'review_date', 'replyContent':'developer_response', 'repliedAt':'developer_response_date'})



In [51]:
df_spotify.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 61690 entries, 0 to 61689
Data columns (total 7 columns):
 #   Column                   Non-Null Count  Dtype         
---  ------                   --------------  -----         
 0   user                     61690 non-null  object        
 1   review_description       61690 non-null  object        
 2   rating                   61690 non-null  int64         
 3   thumbsUpCount            61690 non-null  int64         
 4   review_date              61690 non-null  datetime64[ns]
 5   developer_response       289 non-null    object        
 6   developer_response_date  289 non-null    datetime64[ns]
dtypes: datetime64[ns](2), int64(2), object(3)
memory usage: 3.3+ MB


In [85]:
df_spotify = df_spotify.sort_values(['review_date'], ascending=False, inplace=False)
df_spotify.reset_index(drop=True)

user  \
0      Chethika Abeysinghe   
1             Ric Mcartney   
2           eldorus gilson   
3         Patricia Shelton   
4           Amelia le Roux   
...                    ...   
61685        A Google user   
61686        A Google user   
61687        A Google user   
61688        A Google user   
61689        A Google user   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          review_description  \
0                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                I love this app so much.. It's very good app I recommend this app for use..   
1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             Really like it   
2                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    Easy to work very good!   
3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               Easy to use!   
4                                                                                                                                                                                                                                                                                                                                                                                                                                                                        My bf and I listen to our favourite Scp's on Spotify, It is really worth it, I would recommend it 🥰   
...                                                                                                                                                                                                                                                                                                          

In [53]:
df_spotify.head()

user                                 review_description  \
18379  Chethika Abeysinghe  I love this app so much.. It's very good app I...   
57217         Ric Mcartney                                     Really like it   
43117       eldorus gilson                            Easy to work very good!   
46007     Patricia Shelton                                       Easy to use!   
5302        Amelia le Roux  My bf and I listen to our favourite Scp's on S...   

       rating  thumbsUpCount         review_date developer_response  \
18379       5              0 2022-07-28 13:58:30               None   
57217       5              0 2022-07-27 23:32:52               None   
43117       5              0 2022-07-27 15:29:01               None   
46007       5              0 2022-07-27 15:24:51               None   
5302        5              0 2022-07-27 15:05:26               None   

      developer_response_date  
18379                     NaT  
57217                     NaT  
43117                     NaT  
46007                     NaT  
5302                      NaT

Extract all reviews containing the word design and saving it to a new df

In [94]:
design_word = ["design","Design","button","graphics","ui", "UI", "UX", "ux", "user experience"]
df_spotify_design = df_spotify[df_spotify["review_description"].str.contains("design", "Design")]
df_spotify_design.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 265 entries, 1643 to 21419
Data columns (total 7 columns):
 #   Column                   Non-Null Count  Dtype         
---  ------                   --------------  -----         
 0   user                     265 non-null    object        
 1   review_description       265 non-null    object        
 2   rating                   265 non-null    int64         
 3   thumbsUpCount            265 non-null    int64         
 4   review_date              265 non-null    datetime64[ns]
 5   developer_response       2 non-null      object        
 6   developer_response_date  2 non-null      datetime64[ns]
dtypes: datetime64[ns](2), int64(2), object(3)
memory usage: 16.6+ KB


In [95]:
df_spotify_design.to_csv('spotify_design.csv', index=False)
df = pd.read_csv('spotify_design.csv')

The column review_description is truncated to 50 characters. To inspect some of the reviews fully, I change the display.max_colwidth

In [96]:
pd.set_option('display.max_colwidth', None)

In [105]:
print(df['rating'].value_counts())

5    265
Name: rating, dtype: int64


It seems that all 265 reviews containing the word "design" also gave a 5-Star rating.

## Create a Wordcloud with reviews containing the word Design

In [111]:
from wordcloud import WordCloud, STOPWORDS

In [112]:
review_design = df['review_description']
review_design.to_csv('review_design.csv', index=False)

In [113]:
design_words = open('review_design.csv', mode='r', encoding='utf-8').read()

In [ ]:
stopword_add_ = ""
list_stopadd = stopword_add_.split()
STOPWORDS.update(list_stopadd)

In [115]:
x, y = np.ogrid[:1000, :1000]

mask = (x - 500) ** 2 + (y - 500) ** 2 > 400 ** 2
mask = 255 * mask.astype(int)

wc = WordCloud(
                background_color="white",
                stopwords=STOPWORDS,
                width=600, 
                height=400, 
                mask=mask).generate(design_words)

wc.to_file('design_words.png')